### PR TITLE
coding: Bind to geometry-allocate and attempt to constrain resizes

### PR DESCRIPTION
**Pull request by smspillaz**
This will allow us to keep the minimum size of both windows in
sync.

https://phabricator.endlessm.com/T15968

### CODE REVIEW

File: js/ui/codingManager.js [hunk 1]
```diff
@@ -494,6 +505,11 @@ const CodingSession = new Lang.Class({
             this._sizeChangedIdBuilder = 0;
         }
 
+        if (this._constrainGeometryBuilderId) {
+            this.app.meta_window.disconnect(this._constrainGeometryBuilderId);
```
Comment (cosimoc) [inline js/ui/codingManager.js]: Should be `this.builder.meta_window`

File: js/ui/codingManager.js [hunk 2]
```diff
@@ -533,6 +549,16 @@ const CodingSession = new Lang.Class({
             this._windowUnminimizedId = 0;
         }
 
+        if (this._constrainGeometryAppId) {
+            this.app.meta_window.disconnect(this._constrainGeometryAppId);
+            this._constrainGeometryAppId = 0;
+        }
+
+        if (this._constrainGeometryBuilderId) {
+            this.app.meta_window.disconnect(this._constrainGeometryBuilderId);
```
Comment (cosimoc) [inline js/ui/codingManager.js]: Should be `this.builder.meta_window`

File: js/ui/codingManager.js [hunk 3]
```diff
@@ -561,6 +587,22 @@ const CodingSession = new Lang.Class({
...
+        if (!this.builder)
+            return;
+
+        // Get the minimum size of both the app and the builder window
+        // and then determine the maximum of the two. We won't permit
+        // either window to get any smaller.
+        let [minAppWidth, minAppHeight] = this.app.meta_window.get_minimum_size();
+        let [minBuilderWidth, minBuilderHeight] = this.builder.meta_window.get_minimum_size();
+
+        let minWidth = minAppWidth > minBuilderWidth ? minAppWidth : minBuilderWidth;
```
Comment (cosimoc) [inline js/ui/codingManager.js]: Use `Math.max()`

**Comment by smspillaz** [comment]
Fixed all comments and rebased the branch

**Pull request merged**